# Apply this function to the 'corrected_names' column after running the initial reformatting
temp_study_fix$temp1 <- sapply(temp_study_fix$corrected_names, add_hyphen_after_sci_name, sci_names)

# Starting from here

# Define the output file path
output_file <- "output.txt"

# Open a connection to the file
file_conn <- file(output_file, "w")

# Loop over rows of the study_table
for (i in c(1:nrow(study_table))){
  # Write the index to the file
  cat(paste("Index:", i, "\n"), file=file_conn)
  
  # Write the unique entries corresponding to the current group
  #cat("Unique Entries:\n", file=file_conn)
  cat(unique_entries[groupings == study_table$groupings[i]], sep = ", ", file = file_conn)
  cat("\n\n", file=file_conn)  # Add extra space between entries for clarity
}

# Close the file connection
close(file_conn)

# Confirmation message
cat("Output written to", output_file)


temp_str = unique(temp_study_fix$temp1)
str_c(temp_str, collapse = ",")

"136EKSTRA,16S AMPLIKONS,16S RRNA AMPLICON SEQUENCING FROM VARIOUS SAMPLE TYPES,16S RRNA AMPLICON TESTS,ABL DEEPCHEK,ABL-DEEPCHEK-EVALUATION,ACINETOBACTER LWOFFII F78,AFRICANISOLATES-DZIF,ÄGYPTEN, szulgai-ALERE, kansasii-ALERE,ALEXANDRA DANGEL,ALMATY,AMPLICON,AMPLICON GAUTING,AMPLISEQ,AMPLISEQ TEST,AMPLISEQ-GERBION,ANGEPASSTE TAGMENTIERUNG,ANIMAL-STRAINS-PARIS,ANTWERPEN,ARGENTINA,ARMENIEN,ASERBAIDSCHAN,AUFG-DNA-BELGRADE,AUFTAG NRZ,AUFTRAG,AUFTRAG NRZ,AUFTRAG-KLINIK,AUFTRAG-NGS,AUFTRAG-NRZ,AUFTRÄGE,AZERBAIJAN,BADEN-WÜRTTEMBERG,BADEN-WÜTTEMBERG,BAKU WHO DNA,BAKU WHO LYSATE,BALFS CLINIC,BARBARA HAMMER, basiliense-BASEL,BAYM2015,BDQ MUT,BEIJING,BELGRADE,BELGRADE-LYSATE,BORSTEL MDR (LYSATES-GLYCEROL),BORSTEL PATIENT,BORSTEL-NGSDIAG,BOTSWANA,BOTSWANA GENOSCREEN,BRUKER STUDIE-NRZ,BUKINA FASO, bovis-BUKINA-FASO,BW,Clostridium difficile,CAMEROON-9M,CAP SELEKTIONSKLONE,CARINE-DEL-MUTANT,CAROLIN GOLIN,CD3,CD3-DEEPLEX,CEALAND-TRAINING,CELERO EZ NUQUANT HALF REACTIONS,Stenotrophomonas maltophilia-CF,CHILDHOOD TB, chimaera, chimaera-PACBIO,CHLAMYDIA,Chlamydia pneumoniae-KENSUKE,CHRISTIAN BENZ,CHRONICALLY INFECTED WOMEN,-EEP-STUDIE DARSHA-COBAS ,COLUMBIA-HURTADO,SARS-CoV-2,SARS-CoV-2-KIEL,CROATIA,CRONO CLONE,CRONOCLONE,CRYPTIK,Cupriavidus metallidurans,CUT-TB LESOTHO,CUT-TB SOUTH AFRICA,CUT-TB TANZANIA,CYCLOSERINE IN VITRO MUTANTS,CZ0813-HAEMOPHILUS,DAG,DAG HARMSEN,DEEPLEX,DEEPLEX-AUFTRAG KLINIK,DEEPLEX LOD,DEEPLEX-AUFTRAG,DEEPLEX-BAYM TEST,DEEPLEX-CORVOS-TRAINING,DEEPLEX-HAIN-TEST,DEEPLEX-TRAINING,DEEPLEX-AMPLICON,DELAMANID,DEUSTCHLAND,DEUTSCHLAND,DICK VAN SOOLINGEN,DLM RESISTENZ,DLM-CFZ-BDQ-LZD RESISTANCE MARKER,DNA,DNA EXTRACTION NGS COMPATIBLE?,DNAZYME,DZIF,DZIF-MUT,Escherichia coli,EASTGREENLAND,ECATERINA-PHD,ELEFANT,EMB RIF TP2 1,EMB RIF TP2 2,EMB RIF TP2 3,EMB RIF TP2 4,EMB RIF TP2 5,EMB RIF TP3 1,EMB RIF TP3 2,EMB RIF TP3 3,EMB RIF TP3 4,EMB RIF TP3 5,EMB TP1 1,EMB TP1 2,EMB TP1 3,EMB TP1 4,EMB TP1 5,ERA4TB,ERANET-LAC,ERASE TB,ERASE-TB,ESTONIA,ESWATINI-ITG,ETHIOPIA-JIMMA,EUSEQMYTB,EUSEQMYTB CZECHIA,EUSEQMYTB RUMANIA,EUSEQMYTB RUMANIA BUCHAREST,EUSEQMYTB RUMANIA CLUJ,EXTERNAL ORDER,FEND-TB,FIND,FIND-PERU,FIND-VIETNAM,FIND-DENVER,FIND-GEORGIEN,FIND-MOLDAWIEN,FLI,FLUCTUATION,FLUCTUATION-ISOLATE-SEQ,FLUCTUATION-POP-SEQ,FRANKFURT,FREIBURG,FÜR NGS (SVEN),GA KREIS LOHE NGS,GAMBIA CHILDHOOD TB,GAUTING,GERBION,GHANA,GLENNAH,GREENLAND,H37-KONTROLLE,H37RV,H37RV DOREEN 2010,H37RV-BORSTEL,HAEMO-LUB-RESISTOME,HAEMO-WUR-RESISTOME,HAEMOPHILUS-SERBIA,HAEMOPHILUS-RESISTOME,HAMBURG,HAMBURG EAI,HAMBURG-BADEN-WÜRTEMBERG,HAMBURG-AUFTRAG-NRZ,HAMILTON,HAMILTON AUTOMATIC EXTRACTOR,HAN2000,HAN2006,HAN2012,HAN2013,HANNOVER,HANNOVER-AUFTRAG-KLINIK,HARALD HOFFMANN,HEILBRONN GESUNDHEITSAMT,HERZKLAPPE,HETERORESISTANCE,Haemophilus haemolyticus,HOMBURG,HYS-RESIST CHECK, bovis-IMOSER, microti-IMOSER,INFLUENZA,INTERNAL,IRAK,IRAK-ISSAM,IRAN-MDR,IRMGARD MOSER-FLI,ITALIEN,-ITALIEN,ITM-REFERENCE-COLLECTION,ITS AMPLICON TESTS,Staphylococcus hominis-JÖRG STEINMANN,KASACHSTAN-ARIKE,KASACHSTAN2013,KATHARINA KRANZER,KENIA-GLENNAH,KENIA-PERPETUAL,Chlamydia trachomatis-KENSUKE SHIMA,KENYA,KINFGFISHER ISOLATION,KINGFISHERDUO,KIRIBATI,Klebsiella pneumoniae,KLINIK,KLINIK AUFTRAG,KLINIK-AUFTRAG,KNOBLOCH,KÖLN,KOMMENSALE BAKTERIEN,KONGO,KOSAKONIA,KROATIEN,KYRGYZSTAN,LAMBARENE,LEGIONELLA ANNELIES YINKERNAGEL,LEGIONELLA HUGO SAX,Legionella pneumophila,LEGIONELLA ZÜRICH,LEILA,LGL BAYERN,LIMA,LINEAGE 2 TRANSCRIPTOME,LINEZOLID,LONDON PANEL-NTM-VLAD,LONGDRUG,LOW FREQ,LOW TSCA TEST,LT MUTS,LT-23 MUTANTS,LT22-MUTS,LT23-MUTS,LYSATE, abscessus, abscessus-BORSTEL, abscessus-ESSEN, avium-SCHWESTERN, basiliense, bovis, canettii, chelonae-DIVERSITY, chimaera-DRESDEN, chimaera-HCU (UNI GIEßEN), chimaera-POPULATION, chimaera-WETZSTEIN, chimaera-WAGNER-BRAUN, fortuitum, fortuitum-CONTAMINATION, fortuitum-SEQUENTIAL ISOLATES FROM PATIENT WITH WOUND INFECTION AFTER HEARTH TRANSPLANT, intracellulare, microti-BORSTEL, microti-NGS, smegmatis-TRANSPOSON, smegmatis-MC2 155, szulgai-CHECK,, vulneris-OUTBREAK, avium, avium-BERLIN, bovis-NGS, bovis-BOVIS, bovis-CAPRAE, chimaera-AUFTRAG-KLINIK, chimaera-NGS, chimaera-PROJEKT, chimaera-VARIATION, chimaera-AUFTRAG, chimaera-ZAHNARZT, intracellulare-COMPLEX, kansasii, simiae-LUDWIG SEDLACEK,MABS RNASEQ TEST,MACROPHAGE INFECTION RNA,MAGICPREP TECAN CELERO EZ DNA-SEQ,MAHA FARHAT SOUTH AFRICA,MAILAND,MALAVI RAPAED,MAPUTO,MAPUTO (FÜR MATTHIAS),MARCEL BEHR DEFINED MUTANTS,MDR,MDR COLOMBIA,MDR DEUTSCHLAND,MDR KINSHASA,MDR-TB SEQ NET: ESWATINI,MDR1996,MDR1997,MDR2001,MDR2004,MDR2012,MDR2012-2013,MDR2020-SERBIA,MED TP0 1,MED TP0 2,MED TP0 3,MED TP0 4,MED TP0 5,MED TP1 1,MED TP1 2,MED TP1 3,MED TP1 4,MED TP1 5,MED TP2 1,MED TP2 2,MED TP2 3,MED TP2 4,MED TP2 5,MED TP3 1,MED TP3 2,MED TP3 3,MED TP3 4,MED TP3 5,MFX MUTANTS,MG KOELLSCHEN,MGIT,MGIT-SEDIMENT,MGIT (NRZ),MICHAEL RASMUSSEN,MINION JOHANNA,MINT-ANIMALS,MINT-MVZ-ROSENBOOM,MISCHKULTUR,MOLBIO,MOLDOVA,MOLDOVA-CARE,MOLDOVA-MDR,MOLDOVA-VCM,MOMA CF,MOMA-CF,MOROCCO-HBENHASSOU,MOZAMBIQUE RAPAED,MÜNSTER,MUSIOLIK TRAINING 2023,MUT2003, spp.,N,Neisseria meningitidis,NAMIBIA,NAMIBIA-NIP,NGS ANDERE,NGS LIBRARY KIT COMPARISON,NGS TRAINING,NGS-ANDERE,NGS-ANTWERPEN,NGS-AUFTRAG,NIROS,NORBERT REILING,NRL DEEPLEX,NRZ,NRZ AUFTRAG,NRZ AUFTRAG ARMENIEN,NRZ ROUTINE,NRZ ROUTINE STUDY,NRZ-AUFTRAG,NTM FRANKFURT,NTM PANEL,NTM-FRANKFURT,NTMSCOPE,NUEBEL,NUKUS,NUKUSNEU,NÜRNBERG,OLENA,OPTIMIERUNG 8EURO LIBRARY,OPTIMIZATION DNA EXTRACTION NRZ,OPTIMIZATION OF 16S RRNA AMPLICON SEQUENCING,Orientia tsutsugamushi,OSLO OUTBREAK,Pseudomonas aeruginosa,PA MICS ITM,PAKISTAN2020,PARADIGM4TB,PAS-SELEKTIONSKLON,PATIENT,PATIENT M-XDR DEVELOPMENT,PCR-FREE LIBRARY PREP TEST,Stenotrophomonas maltophilia-PEG,PERU,PHAGENPROJEKT,PHD WORK,PILOT STUDY PAKISTAN,PKNL MUTANTS,PLASMIDS NTM,PLATE-1-BEIJING 100-32,PLATE-2-BEIJING 100-32,PLATE-3-94-32,PMA-TEST,PRETOMANID,PRETONAMID, kansasii-PS,QUALITY CONTROL PANEL,QUALITY-STRAIN,QUICK GENE MINI ISOLATION,R 01 TB BOTSWANA-SPUTUM-DNA,RAPAED,RAPAED-STUDIE,RAPEAD,RASMUSSEN,RECURRENT TB,REF. KO,REF. KOL,REFERENCE,REFERENZKOLL,REFERENZKOLLEKTION,REILING-H37RV,RESI,RESI2001,RESI2005,RESI2006,RESI2007,RESI2009,RESI2011,RESI2012,RESI2013,RESISTOME HAEMOPHILUS STRAINS,RESISTOME STRAINS,REVELO DNA-SEQ ENZ FOR MAGICPREP,Rickettsia typhi,RIF,RIF EMB TP2 1,RIF EMB TP2 2,RIF EMB TP2 3,RIF EMB TP2 4,RIF EMB TP2 5,RIF EMB TP3 1,RIF EMB TP3 2,RIF EMB TP3 3,RIF EMB TP3 4,RIF EMB TP3 5,RIF TP1 1,RIF TP1 2,RIF TP1 3,RIF TP1 4,RIF TP1 5,RING TRIAL,RINGVERSUCH,RNAS SVEN,ROMANIA-MNI,RUMÄNIEN,RUMÄNIEN NGS,S,S. AFRICA INH MUTANT,Staphylococcus aureus,Staphylococcus epidermidis,Stenotrophomonas maltophilia,Stenotrophomonas maltophilia-EXPERIMENTAL EVOLUTION IN IN VITRO LUNG EPITHELIUM BIOFILM MODEL,Stenotrophomonas maltophilia-PATIENTS,Stenotrophomonas acidaminiphila,S. humi,Sphingomonas koreensis,Stenotrophomonas nitritireducens,Stenotrophomonas rhizophila,S. terrae,S3 STRAINS,SAKAI,SAMBIA,SARAH LEPUSCHITZ,SARAH SENGSTAKE,SAUDI-ARABIEN,SCHLESWIG-HOLSTEIN,SCHORPP FÜR LINDSEY,SEDIMENT-MGIT,SEKRETOM,SELEKTIONSKLON P22-42,SELEKTIONSKLON SILKE,SELEKTIONSKLONE,SELEKTIONSKLONE SILKE,SENEGAL,SENEGAL-AWA,SEQMDRTB-NET,SEQMDRTB-NET:MOZAMBIQUE,SERBIA,SERRATIA,SH,SHOTGUN MATTHIAS,SHOTGUN METAGENOM,SHOTGUN METAGENOMICS FROM LOW BIOMSS SAMPLES,SIERRA LEONE,SL-FIND,SOUTH AFRICA,SPECIES CHECK,SPEICHEL BEADS,SPUTUM,SPUTUM NEB ENRICHMENT,SPUTUM PILOT STUDY,STÄMME DIAGNOSTIK LÜBECK,STAPHYLOCOCCUS EPIDERMIDIS,STEHEN IM MISEQ-FACH,Stenotrophomonas spp.,Stenotrophomonas maltophilia-K279A,STOOL DIAGNOSTIC,STOOL SAMPLES,STOPTTH,STUDIE BCD,STUHL PROBEN KLINIK,STUTTGART MUT,SUDAN,SUDAN-YASSIR,SUSANNE 49-02,SWABS-BOTSWANA2023, SWAZILAND,TANSANIA,TANZANIA,TANZANIA-BMUTAYOBA,TB PORTAL,TB PORTAL GEORGIA,TB PORTAL MEXICO,TB PORTAL NIGERIA,TB PORTAL ROMANIA,TB SEQDISK,TB SEQUEL SÜD AFRIKA,TB SEQUEL TANZANIA,TB-AMAZON COLOMBIA,TB-PORTAL BELARUS,TB-PORTAL CONGO,TB-PORTAL GEORGIA,TB-PORTAL GEORGIEN,TB-PORTAL KAZAKHSTAN,TB-PORTAL MEXICO,TB-PORTAL MOLDOVA,TB-PORTAL NIGERIA,TB-PORTAL ROMANIA,TB-PORTAL UKRAINE,TB-SEQDISK,TB-SEQDISK ENRICHMENT FACTOR,TB-SEQUEL GAMBIA,TB-SEQUEL MOSAMBIK,TB-SEQUEL SOUTHAFRICA,TB-SEQUEL TANZANIA,TB-SEQUEL-SOUTH-AFRICA,TECAN CELERO EZ DNA-SEQ LIBRARY PREP,TECAN MAGICPREP NGS LIBRARY,TEST 8EURO LIBRARY,THAILAND,THOMAS,TIERPATHOGENE,TMW-DC-MUT-M3,TMW-H37-RR-CHECK,TRAINING,TRAINING QATAR AND OLGA (DEEPLEX),TRAINING QATAR AND OLGA (LT MUTS),TRAINING QATAR AND OLGA (QUALITY CONTROL),TSCA TEST,TSCA TRAINING,TUNESIEN,TUNISIA,TURK,TURKMENISTAN,TURMENISTAN,TW-DC-MUT,TW-LS-01-BDQ-MUT,TYWA,ÜBERSTAND-MGIT,ÜBERTRAGUNGSNACHWEIS,UKRAINE,UKRAINE2020,UKRAINE2021,UKRAINE2022,UKRAINE2023,-UKRAINIAN ,UNKNOWN PROJECT DORIS HILLEMANN,UZBEKISTAN,UZL-RECTAL-SWABS2023,VIDHISHA-MUTANTS,VIOLA MIXE,VIRUS DETECTION,VORTEX-ÜBERST-MGIT,W148,WHO PANEL ANTWEERPEN,WHO PANEL ANTWERPEN,WIEN,WP2 WP4,XDR,XDR-NAMIBIA,YASSIR,ZYMO-SEQ"

# Apply the replacement function to the column
# New
temp_study_fix$temp3 <- replace_with_map(temp_study_fix$temp1, country_map)

# Old
temp_study_fix$temp2 <- sapply(temp_study_fix$temp1, replace_country_city, country_map)


# View the updated table
print(temp_source_tab)




